addons/gamification/models/gamification_karma_tracking.py

```diff
@@ -1,69 +1,133 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-import calendar
-
 from dateutil.relativedelta import relativedelta
 
-from odoo import api, fields, models
+from odoo import _, api, fields, models
+from odoo.tools import date_utils
 
 
 class KarmaTracking(models.Model):
     _name = 'gamification.karma.tracking'
     _description = 'Track Karma Changes'
     _rec_name = 'user_id'
-    _order = 'tracking_date DESC'
+    _order = 'tracking_date desc, id desc'
+
+    def _get_origin_selection_values(self):
+        return [('res.users', 'User')]
 
-    user_id = fields.Many2one('res.users', 'User', index=True, readonly=True, required=True, ondelete='cascade')
-    old_value = fields.Integer('Old Karma Value', required=True, readonly=True)
-    new_value = fields.Integer('New Karma Value', required=True, readonly=True)
+    user_id = fields.Many2one('res.users', 'User', index=True, ondelete='cascade')
+    old_value = fields.Integer('Old Karma Value', readonly=True)
+    new_value = fields.Integer('New Karma Value', required=True)
+    gain = fields.Integer('Gain', compute='_compute_gain', readonly=False)
     consolidated = fields.Boolean('Consolidated')
-    tracking_date = fields.Date(default=fields.Date.context_today)
+
+    tracking_date = fields.Datetime(default=fields.Datetime.now, readonly=True, index=True)
+    reason = fields.Text(default=lambda self: _('Added Manually'), string='Description')
+    origin_ref = fields.Reference(
+        string='Source',
+        selection=lambda self: self._get_origin_selection_values(),
+        default=lambda self: f'res.users,{self.env.user.id}',
+    )
+    origin_ref_model_id = fields.Many2one(
```

Comment:
People won't have access to it ... so not really usable I fear. ir.model is not something we want to use (well I want but we are not allowed :( )